framework/core/src/Filter/ValidateFilterTrait.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Filter;
+
+use Flarum\Foundation\ValidationException as FlarumValidationException;
+use Flarum\Locale\Translator;
+
+trait ValidateFilterTrait
+{
+    /**
+     * @throws FlarumValidationException
+     */
+    protected function asArray($filterValue, $multidimensional = false): array
```

Comment:
Should these require `string`? / why not add type annotations?

Reply:
`filterValue` is either a string or an array, w can't use `mixed` and we can't use `sting|array` yet.

The second parameter does need to be typed though.